### ISSUE TITLE
Unable to recover from fork

### ISSUE THREAD

**Issue by jplanc** [EOSIO/demux-js-postgres#23]
Demux encountered a small fork in a testnet. It detected the fork but was unable to recover.

```
demux-0  | error retrieving block, retrying...
demux-0  | !! FORK DETECTED !!
demux-0  |   MISMATCH:
demux-0  |     ✓ NEW Block 2273566 previous: 0022b11d2f62c2b73bafb7d3d5c06a4e667b61bbc14c12fda61c2b07b4089037
demux-0  |     ✕ OLD Block 2273565 id:       0022b11d6f5bdc839a75918a388750bd621bf904d74d39465e71a8464fb21312
demux-0  | Refetching Block 2273565...
demux-0  |   MATCH:
demux-0  |     ✓ NEW Block 2273565 previous: 0022b11c4ca1a958c7db389fa939e225afa7acfa211f61cf43eed17208d592e7
demux-0  |     ✓ OLD Block 2273564 id:       0022b11c4ca1a958c7db389fa939e225afa7acfa211f61cf43eed17208d592e7
demux-0  | !! FORK RESOLVED !!
demux-0  | Rolling back 1 blocks to block 2273564...
demux-0  | ROLLING BACK BLOCK 2273565
demux-0  | You have triggered an unhandledRejection, you may have forgotten to catch a Promise rejection:
demux-0  | TypeError: Cannot read property 'fn_undo_transaction' of undefined
demux-0  |     at MassiveActionHandler.<anonymous> (/opt/eosio-vault-transfer/services/transfer/demux/node_modules/demux-postgres/dist/MassiveActionHandler.js:92:54)
demux-0  |     at Generator.next (<anonymous>)
demux-0  |     at fulfilled (/opt/eosio-vault-transfer/services/transfer/demux/node_modules/demux-postgres/dist/MassiveActionHandler.js:4:58)
demux-0  |     at process._tickCallback (internal/process/next_tick.js:68:7)
[rundev] App demux restarted
demux-0  | error retrieving block, retrying...
demux-0  | error retrieving block, retrying...
demux-0  | error retrieving block, retrying...
demux-0  | error retrieving block, retrying...
demux-0  | error getting head block number, retrying...
```

Postgres is stuck at
```
# select * from transfer._index_state;
 id | block_number |                            block_hash                            | is_replay 
----+--------------+------------------------------------------------------------------+-----------
  1 |      2273565 | 0022b11d6f5bdc839a75918a388750bd621bf904d74d39465e71a8464fb21312 | f
```

**Comment by jeffreyssmith2nd** [comment]
Talked with @jplanc in person about this, will be taking a look today

**Comment by jeffreyssmith2nd** [comment]
I expect this issue is somewhat related to #24

**Comment by flux627** [comment]
Most likely resolved via #25. Feel free to re-open if that's not the case.

**Comment by jeffreyssmith2nd** [comment]
The full resolution to this issue is also to use the built in migration system in demux rather than the `migrate.sh` system that we've been temporarily using.
